src/java/org/apache/cassandra/repair/autorepair/IAutoRepairTokenRangeSplitter.java

```diff
@@ -19,13 +19,83 @@
 
 
 import java.util.List;
+import java.util.Objects;
+import java.util.Set;
 
+import org.apache.cassandra.dht.Range;
 import org.apache.cassandra.dht.Token;
 import org.apache.cassandra.utils.Pair;
 
 public interface IAutoRepairTokenRangeSplitter
 {
-    // split the token range you wish to repair into multiple subranges
-    // the autorepair framework will repair the list of returned subrange in a sequence
-    List<Pair<Token, Token>> getRange(AutoRepairConfig.RepairType repairType, boolean primaryRangeOnly, String keyspaceName, String tableName);
+
+    /**
+     * Split the token range you wish to repair into multiple assignments.
+     * The autorepair framework will repair the list of returned subrange in a sequence.
+     * @param repairType The type of repair being executed
+     * @param primaryRangeOnly Whether to repair only this node's primary ranges or all of its ranges.
+     * @param keyspaceName The keyspace being repaired
+     * @param tableNames The tables to repair
+     * @return repair assignments broken up by range, keyspace and tables.
+     */
+    List<RepairAssignment> getRepairAssignments(AutoRepairConfig.RepairType repairType, boolean primaryRangeOnly, String keyspaceName, Set<String> tableNames);
```

Comment:
Main API change.  Thoughts?   The naming could change, but didn't want to think so much about that, rather just have something we can chat about.